From ab0d22fe74dc90c13ea4c94c48fdb0d1a9730b94 Mon Sep 17 00:00:00 2001
From: Manuel Moos <z-man@users.sf.net>
Date: Sun, 2 Mar 2025 00:04:32 +0100
Subject: [PATCH] Adapt to future changed return code of nKrawall::FetchURL

More fully map actual URI fetching return codes to FetchURL return codes
---
 src/network/nAuthentication.cpp |  4 ++--
 src/network/nKrawallPrivate.cpp | 13 ++++---------
 2 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/network/nAuthentication.cpp b/src/network/nAuthentication.cpp
index 0e3625c24..16b0028aa 100644
--- a/src/network/nAuthentication.cpp
+++ b/src/network/nAuthentication.cpp
@@ -788,7 +788,7 @@ bool nLoginProcess::FetchInfoFromAuthorityRemote()
         std::stringstream answer;
         int rc = nKrawall::FetchURL( fullAuthority, "?query=methods", answer );
 
-        if ( rc == -1 )
+        if (rc < 0)
         {
             return ReportAuthorityError( tOutput( "$login_error_invalidurl_notfound", authority ) );
         }
@@ -835,7 +835,7 @@ bool nLoginProcess::FetchInfoFromAuthorityRemote()
         
         if ( rc != 200 )
         {
-            if ( rc == -1 )
+            if (rc < 0)
             {
                 return ReportAuthorityError( tOutput( "$login_error_invalidurl_notfound", authority ) );
             }
diff --git a/src/network/nKrawallPrivate.cpp b/src/network/nKrawallPrivate.cpp
index a0d6a37e8..fd717d801 100644
--- a/src/network/nKrawallPrivate.cpp
+++ b/src/network/nKrawallPrivate.cpp
@@ -334,7 +334,7 @@ void nKrawall::CheckScrambledPassword( nCheckResultBase & result,
         std::stringstream content;
         int rc = FetchURL( data.fullAuthority, request.str().c_str(), content );
 
-        if (rc == -1)
+        if (rc < 0)
         {
             result.error = tOutput( "$login_error_invalidurl_notfound", result.authority );
             result.success = false;
@@ -459,15 +459,10 @@ int nKrawall::FetchURL(tString const& authority, char const* query, std::ostream
     // con << "Fetching authentication URL " << fullURL.str() << "\n";
 
     // fetch URL
-    tResourceManager::Result ret = tResourceManager::FetchURI(fullURL.str().c_str(), target);
-
-    switch (ret)
-    {
-    case tResourceManager::Result::OK:
-        return 200;
-    default:
+    int ret = tResourceManager::FetchURI(fullURL.str().c_str(), target);
+    if (ret < 0)
         return -1;
-    }
+    return ret;
 }
 
 #ifdef KRAWALL_SERVER_LEAGUE
-- 
GitLab

